### PR TITLE
Rename canvas module, add skin + background change

### PR DESCRIPTION
**Pull request by tjorim**
## Description

Rename canvas to canvasUtil as agreed in #248
Add option to allow users changing the background manually #227
Re-added the skin change options, can still be removed #213
Apply previous fixes:
- use other method to maintain zoom
- simplify the mobilerendering again (and add back the black background for now)
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by tjorim** [comment]
Don't merge yet, I messed something up with the mouselistener.

File: bot.user.js [hunk 1]
```diff
@@ -1107,7 +1166,7 @@ var userInterface = window.userInterface = (function () {
                     case 1:
                         bot.defaultAccel = 1;
                         if (!bot.isBotEnabled) {
-                            original_onmouseDown(e);
```
Comment (ermiyaeskandary) [inline bot.user.js:1110]: This is where it is starting to get messy...

File: bot.user.js [hunk 2]
```diff
@@ -1121,24 +1180,12 @@ var userInterface = window.userInterface = (function () {
             userInterface.onPrefChange();
         },
 
-        onmouseup: function () {
```
Comment (ermiyaeskandary) [inline bot.user.js:1124]: Why have you removed these ? :confused:

File: bot.user.js [hunk 3]
```diff
@@ -1038,19 +1074,35 @@ var userInterface = window.userInterface = (function () {
...
                 // Letter 'I' to toggle autorespawn
                 if (e.keyCode === 73) {
                     window.autoRespawn = !window.autoRespawn;
                     window.log('Automatic Respawning set to: ' + window.autoRespawn);
                     userInterface.savePreference('autoRespawn', window.autoRespawn);
                 }
-                // Letter 'H' to toggle hidden mode
-                if (e.keyCode === 72) {
-                    userInterface.toggleOverlays();
+                // Letter 'W' to auto rotate skin
```
Comment (ermiyaeskandary) [inline bot.user.js:1087]: Apparently they are client side as I have never seen anyone with rotating skin and IMO it is just extra client side code, putting extra burden on the CPU...

**Comment by ermiyaeskandary** [comment]
Lemme do the the CanvasUtil and then you can do seperate PRs for the skin change and the background so everything is nice and neat :+1:

**Comment by ermiyaeskandary** [comment]
The background thing looks cool :+1:  - did it :+1:

**Comment by ChadSki** [comment]
@tjorim is this still in progress?

**Comment by ermiyaeskandary** [comment]
No point anymore anyway. Background change and renaming has been done and skin change is no longer wanted.

**Pull request closed without merging**